src/util/ajax.js

```diff
@@ -158,7 +161,22 @@ function makeXMLHttpRequest(requestParameters: RequestParameters, callback: Resp
     return { cancel: () => xhr.abort() };
 }
 
-const makeRequest = window.fetch && window.Request && window.AbortController ? makeFetchRequest : makeXMLHttpRequest;
+export const makeRequest = function(requestParameters: RequestParameters, callback: ResponseCallback<any>): Cancelable {
+    // We're trying to use the Fetch API if possible. However, in some situations we can't use it:
+    // - IE11 doesn't support it at all. In this case, we dispatch the request to the main thread so
+    //   that we can get an accruate referrer header.
+    // - Requests for resources with the file:// URI scheme don't work with the Fetch API either. In
+    //   this case we unconditionally use XHR on the current thread since referrers don't matter.
+    if (!/^file:/.test(requestParameters.url)) {
```

Comment:
Doesn't work with relative URLs

Reply:
I can confirm this, you have to specify the absolute path with the file protocol before passing it in, else it thinks it is an http request and throws an error about the file protocol like before.

Reply:
We don't support relative URLs intentionally — see the discussion at #3636 and linked tickets. That said,  https://github.com/mapbox/mapbox-gl-js/pull/8612 should now correctly classify these URLs as file URLs.

Reply:
We don't support relative URLs intentionally — see the discussion at #3636 and linked tickets. That said, #8612 should now correctly classify these URLs as file URLs.